cql3/expr/prepare_expr.cc

```diff
@@ -24,6 +24,25 @@
 
 namespace cql3::expr {
 
+static
+lw_shared_ptr<column_specification>
+column_specification_of(const expression& e) {
+    return visit(overloaded_functor{
+        [] (const column_value& cv) {
+            return cv.col->column_specification;
+        },
+        [&] (const ExpressionElement auto& other) {
+            auto type = type_of(e);
+            if (!type) {
+                throw exceptions::invalid_request_exception(fmt::format("cannot infer type of {}", e));
+            }
+            // fake out a column_identifier
```

Comment:
Let's add an explicit FIXME comment here, explaining why it's bad, why it's good enough for now, and perhaps suggesting a follow-up fix.

Reply:
Done